opcua/common/event.py

```diff
@@ -22,53 +22,113 @@ class Event(object):
-        self.EventId = uuid.uuid4().bytes
-        self.EventType = self.node.nodeid
-        self.LocaleTime = datetime.utcnow()
-        self.ReceiveTime = datetime.utcnow()
-        self.Time = datetime.utcnow()
-        self.Message = ua.LocalizedText()
-        self.Severity = ua.Variant(1, ua.VariantType.UInt16)
-        self.SourceName = "Server"
-
-        # og set some node attributed we also are expected to have
-        self.BrowseName = self.node.get_browse_name()
-        self.DisplayName = self.node.get_display_name()
-        self.NodeId = self.node.nodeid
-        self.NodeClass = self.node.get_node_class()
-        self.Description = self.node.get_description()
+            source = Node(isession, ua.NodeId(source))
+
+        if self.event.SourceNode.Identifier:
+            source = Node(self.isession, self.event.SourceNode)
+
+        self.event.SourceName = source.get_display_name().Text
+        source.set_attribute(ua.AttributeIds.EventNotifier, ua.DataValue(ua.Variant(1, ua.VariantType.Byte)))
 
     def __str__(self):
-        return "Event(Type:{}, Source:{}, Time:{}, Message: {})".format(self.EventType, self.SourceNode, self.Time, self.Message)
+        return "EventGenerator(Type:{}, Source:{}, Time:{}, Message: {})".format(self.EventType, self.SourceNode, self.Time, self.Message)
     __repr__ = __str__
 
-    def trigger(self):
+    def trigger(self, time=None, message=None):
         """
         Trigger the event. This will send a notification to all subscribed clients
         """
-        self.isession.subscription_service.trigger_event(self)
-
-    def _set_members_from_node(self, node):
-        references = node.get_children_descriptions(refs=ua.ObjectIds.HasProperty)
-        for desc in references:
-            node = Node(self.isession, desc.NodeId)
-            setattr(self, desc.BrowseName.Name, node.get_value())
+        self.event.EventId = ua.Variant(uuid.uuid4().hex, ua.VariantType.ByteString)
+        if time:
+            self.event.Time = time
+        else:
+            self.event.Time = datetime.utcnow()
+        self.event.RecieveTime = datetime.utcnow()
+        #FIXME: LocalTime is wrong but currently know better. For description s. Part 5 page 18
+        self.event.LocalTime = datetime.utcnow()
+        if message:
+            self.Message = ua.LocalizedText(message)
+        elif not self.event.Message:
+            self.event.Message = ua.LocalizedText(self.source.get_browse_name().Text)
+        self.isession.subscription_service.trigger_event(self.event)
+
+
+def get_event_from_node(node):
+    if node.nodeid.Identifier in ua.uaevents_auto.IMPLEMNTED_EVENTS.keys():
+        return ua.uaevents_auto.IMPLEMNTED_EVENTS[node.nodeid.Identifier]()
+    else:
+        parent_identifier, parent_eventtype = _find_parent_eventtype(node)
+        if not parent_eventtype:
+            return None
+
+        class CustomEvent(parent_eventtype):
```

Comment:
We do not need to have this as an internal class. We can as well call setattr after creating the object, or give everything necessary as arg to __init__ og CustomEvent